src/utils_single_edge.jl

```diff
@@ -0,0 +1,216 @@
+function KLIEP_debias(
+    ind::Int64,
+    θ,
+    ω::SparseVector,
+    Ψx::Matrix{Float64},
+    Ψy::Matrix{Float64},)
+
+    μx = vec(mean(Ψx, dims=2))
+    ry = zeros(size(Ψy,2))
+    mul!(ry, transpose(Ψy), θ)
+
+    θ1 = θ[ind]
+    for k in ω.nzind
+        θ1 += ω[k] * (μx[k] - mean( exp.(ry) .* Ψy[k, :] ) / mean(exp, ry))
```

Comment:
You are recomputing 
`exp.(ry)
`and
`mean(exp, ry)`
in each iteration even though they do not change from iteration to iteration.

Reply:
Fixed!